### PR TITLE
Feat: implements class relationships

### PR DESCRIPTION
**Pull request by pabloalastomas**

### CODE REVIEW

File: src/main/java/cat/udl/eps/softarch/demo/domain/Request.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 public class Request extends Batch {
     @Id
     private Long id;
-
     private ZonedDateTime lastUpdate;
+    @OneToOne
+    private Take take;
```
Comment (rogargon) [inline src/main/java/cat/udl/eps/softarch/demo/domain/Request.java:21]: Better to use the name we have in the entity diagram "takenBy"

File: src/main/java/cat/udl/eps/softarch/demo/domain/Take.java [hunk 2]
```diff
@@ -17,6 +17,10 @@ public class Take extends Batch {
     @Id
     private Long id;
     private ZonedDateTime lastUpdate;
-
+    @OneToOne
+    private Request request;
```
Comment (rogargon) [inline src/main/java/cat/udl/eps/softarch/demo/domain/Take.java:21]: No need to model both sides of the relationship, as per de diagram we are currently modelling the other side. If we model both, we will need to take care of having both in sync

File: src/main/java/cat/udl/eps/softarch/demo/domain/Request.java (hunk 1 above)
Comment (pabloalastomas) [reply src/main/java/cat/udl/eps/softarch/demo/domain/Request.java]: I assume that in this case you mean fulfilled.

File: src/main/java/cat/udl/eps/softarch/demo/domain/Take.java (hunk 2 above)
Comment (pabloalastomas) [reply src/main/java/cat/udl/eps/softarch/demo/domain/Take.java]: Perfect, I take it into account.

**Review by rogargon — APPROVED** [review]

**Pull request merged**